addons/mail/static/src/discuss/call/common/call.js

```diff
@@ -109,17 +132,54 @@ export class Call extends Component {
     }
 
     get visibleMainCards() {
-        if (this.props.thread.activeRtcSession) {
+        const activeSession = this.props.thread.activeRtcSession;
+        if (activeSession) {
+            let videoType;
+            switch (activeSession.videoStreams.size) {
+                case 1:
+                    this.state.insetCard = false;
+                    if (
+                        activeSession.mainVideoStream !==
+                        activeSession.videoStreams.values().next().value
+                    ) {
+                        this.props.thread.activeRtcSession = false;
+                        activeSession.mainVideoStream = false;
+                    }
+                    break;
+                case 2:
+                    videoType =
+                        activeSession.videoStreams.get("screen") === activeSession.mainVideoStream
+                            ? "camera"
+                            : "screen";
+                    this.setInset(activeSession, videoType);
+                    break;
+            }
+
             return [
                 {
-                    key: "session_" + this.props.thread.activeRtcSession.id,
-                    session: this.props.thread.activeRtcSession,
+                    key: "session_" + activeSession.id,
```

Comment:
Looks like this pattern `"session_" + activeSession.id` is used quite often. How about creating a getter for this on RtcSession?